File: flink_java/src/main/java/github/yahuili1128/streamOperators/ConnectDemo.java
package github.yahuili1128.streamOperators;

import github.yahuili1128.connector.SourceKafka010;
import github.yahuili1128.pojo.MockUpModel;
import github.yahuili1128.pojo.Person;
import org.apache.flink.streaming.api.datastream.ConnectedStreams;
import org.apache.flink.streaming.api.datastream.SingleOutputStreamOperator;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
import org.apache.flink.streaming.api.functions.co.CoMapFunction;

/**
 * @Description :  flink connect java
 * @Author : LiYahui
 * @Date : 2019-07-15 13:02
 * @Version : V1.0
 */
public class ConnectDemo {
	public static void main(String[] args) throws Exception {
		final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
		SingleOutputStreamOperator<MockUpModel> mockUpkafka010 = SourceKafka010.getMockUpkafka010(env);
		SingleOutputStreamOperator<Person> personKafka010 = SourceKafka010.getPersonKafka010(env);
		ConnectedStreams<MockUpModel, Person> resDS = mockUpkafka010.connect(personKafka010);
		// 接下来的操作待定，还不知道怎么去处理



		env.execute("kafka010 demo");
	}
}

File: flink_scala/src/main/scala/github/yahuili1128/streamOperators/ConnectDemo.scala
package github.yahuili1128.streamOperators

import github.yahuili1128.connector.ScalaSourceKafka010
import org.apache.flink.streaming.api.scala.{ConnectedStreams, DataStream, StreamExecutionEnvironment}

/**
  * @Description : flink connect demo
  * @Author : LiYahui
  * @Date : 2019-07-15 18:24
  * @Version : V1.0
  */
object ConnectDemo {
	def main(args: Array[String]): Unit = {
		val env: StreamExecutionEnvironment = StreamExecutionEnvironment.getExecutionEnvironment
		val mockup: DataStream[ScalaSourceKafka010.MockUp] = ScalaSourceKafka010.getMockUpFromKafka(env)
		val personDS: DataStream[ScalaSourceKafka010.Person] = ScalaSourceKafka010.getPersonFromKafka(env)
		val connectedDS: ConnectedStreams[ScalaSourceKafka010.MockUp, ScalaSourceKafka010.Person] = mockup.connect(personDS)
		// 暂时放缓，不知道如何处理


		env.execute(s"${this.getClass.getSimpleName}")
	}

}
